modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/PagedReflectiveOperationExecutorFactory.java

```diff
@@ -0,0 +1,95 @@
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.execution;
+
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+import static org.mule.runtime.module.extension.api.util.MuleExtensionUtils.getInitialiserEvent;
+import static org.mule.runtime.module.extension.internal.runtime.resolver.ValueResolvingContext.from;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.meta.model.ComponentModel;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.streaming.iterator.ConsumerStreamingIterator;
+import org.mule.runtime.core.api.streaming.iterator.ListConsumer;
+import org.mule.runtime.core.privileged.event.BaseEventContext;
+import org.mule.runtime.extension.api.runtime.operation.ComponentExecutor;
+import org.mule.runtime.extension.api.runtime.operation.ComponentExecutorFactory;
+import org.mule.runtime.extension.api.runtime.operation.ExecutionContext;
+import org.mule.runtime.extension.api.runtime.streaming.PagingProvider;
+import org.mule.runtime.module.extension.api.runtime.privileged.ExecutionContextAdapter;
+import org.mule.runtime.module.extension.internal.runtime.connectivity.ExtensionConnectionSupplier;
+import org.mule.runtime.module.extension.internal.runtime.objectbuilder.DefaultObjectBuilder;
+import org.mule.runtime.module.extension.internal.runtime.operation.ReflectiveMethodOperationExecutor;
+import org.mule.runtime.module.extension.internal.runtime.resolver.StaticValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.streaming.PagingProviderProducer;
+import org.mule.runtime.module.extension.internal.util.ReflectionCache;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+import javax.inject.Inject;
+
+/**
+ * An implementation of {@link ComponentExecutorFactory} which produces instances of {@link PagedReflectiveOperationExecutorFactory}.
+ *
+ * @param <T> the type of the class in which the implementing method is declared
+ * @since 4.1.6,4.2.0
+ */
+public final class PagedReflectiveOperationExecutorFactory<T, M extends ComponentModel> implements ComponentExecutorFactory<M> {
+
+  private final Class<T> implementationClass;
+  private final Method operationMethod;
+  private final PagingProviderResultTransformer pagingProviderTransformer = new PagingProviderResultTransformer();
+
+  public PagedReflectiveOperationExecutorFactory(Class<T> implementationClass, Method operationMethod) {
+    checkArgument(implementationClass != null, "implementationClass cannot be null");
+    checkArgument(operationMethod != null, "operationMethod cannot be null");
+
+    this.implementationClass = implementationClass;
+    this.operationMethod = operationMethod;
+
+  }
+
+  @Override
+  public ComponentExecutor<M> createExecutor(M operationModel, Map<String, Object> parameters) {
```

Comment:
bunch of repeated code with ReflectiveOperationExecutorFactory, move to superclass or some other component.

Reply:
fixed